src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/CollectMultipleMetricsSpark.java

```diff
@@ -0,0 +1,181 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines.metrics;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.metrics.Header;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.metrics.MetricAccumulationLevel;
+import org.broadinstitute.hellbender.metrics.QualityYieldMetricsArgs;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.util.*;
+import java.util.function.Consumer;
+
+/**
+ * Tool that instantiates and executes multiple metrics programs using a single RDD.
+ */
+@CommandLineProgramProperties(
+        summary = "Takes an input SAM/BAM/CRAM file and reference sequence and runs one or more " +
+                "metrics modules at the same time to cut down on I/O. Currently all programs are run with " +
+                "default options and fixed output extensions, but this may become more flexible in future.",
+        oneLineSummary = "A \"meta-metrics\" calculating program that produces multiple metrics for the provided SAM/BAM/CRAM file",
+        programGroup = SparkProgramGroup.class
+)
+public final class CollectMultipleMetricsSpark extends GATKSparkTool {
+
+    private static final long serialVersionUID = 1L;
+
+    @Argument(fullName=StandardArgumentDefinitions.ASSUME_SORTED_LONG_NAME,
+                shortName = StandardArgumentDefinitions.ASSUME_SORTED_SHORT_NAME,
+                doc = "If true (default), then the sort order in the header file will be ignored.")
+    public boolean ASSUME_SORTED = true;
+
+    //TODO: do we need this for Spark ?
+    //@Argument(doc = "Stop after processing N reads, mainly for debugging.")
+    //public int STOP_AFTER = 0;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+                shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+                doc = "Base name of output files.")
+    public String outputBaseName;
+
+    @Argument(shortName="LEVEL", doc="The level(s) at which to accumulate metrics.  ")
+    public Set<MetricAccumulationLevel> metricAccumulationLevel = EnumSet.of(MetricAccumulationLevel.ALL_READS);
+
+    @Argument(doc = "List of metrics collectors to apply during the pass through the SAM file.")
+    public List<Collectors> collectors = new ArrayList<>();
+
+    private interface CollectorPrimer {
```

Comment:
I wanted something to indicate the main thing these do, which is to populate the collector with its input arguments. Sounds like primer doesn't do that; provider is, as you say, probably more idiomatic